File: 2026/08-August/05-Wednesday/solution.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public IList<int> RemainingMethods(int n, int k, int[][] invocations)
    {
        List<int>[] graph = new List<int>[n];
        List<int>[] undirected = new List<int>[n];

        for (int i = 0; i < n; i++)
        {
            graph[i] = new List<int>();
            undirected[i] = new List<int>();
        }

        foreach (var edge in invocations)
        {
            int u = edge[0];
            int v = edge[1];

            graph[u].Add(v);
            undirected[u].Add(v);
            undirected[v].Add(u);
        }

        bool[] suspicious = new bool[n];
        DFS(k, graph, suspicious);

        bool[] visited = new bool[n];

        for (int i = 0; i < n; i++)
        {
            if (!suspicious[i] && !visited[i])
            {
                DFS2(i, undirected, suspicious, visited);
            }
        }

        List<int> ans = new List<int>();

        for (int i = 0; i < n; i++)
        {
            if (!suspicious[i])
                ans.Add(i);
        }

        return ans;
    }

    private void DFS(int node, List<int>[] graph, bool[] suspicious)
    {
        suspicious[node] = true;

        foreach (int next in graph[node])
        {
            if (!suspicious[next])
                DFS(next, graph, suspicious);
        }
    }

    private void DFS2(int node, List<int>[] graph, bool[] suspicious, bool[] visited)
    {
        visited[node] = true;

        foreach (int next in graph[node])
        {
            if (!visited[next])
            {
                suspicious[next] = false;
                DFS2(next, graph, suspicious, visited);
            }
        }
    }
}

File: 2026/08-August/05-Wednesday/solution.go
func remainingMethods(n int, k int, invocations [][]int) []int {
	graph := make([][]int, n)
	undirected := make([][]int, n)

	for _, e := range invocations {
		u, v := e[0], e[1]
		graph[u] = append(graph[u], v)
		undirected[u] = append(undirected[u], v)
		undirected[v] = append(undirected[v], u)
	}

	suspicious := make([]bool, n)

	var dfs func(int)
	dfs = func(u int) {
		suspicious[u] = true
		for _, v := range graph[u] {
			if !suspicious[v] {
				dfs(v)
			}
		}
	}

	dfs(k)

	visited := make([]bool, n)

	var dfs2 func(int)
	dfs2 = func(u int) {
		visited[u] = true
		for _, v := range undirected[u] {
			if !visited[v] {
				suspicious[v] = false
				dfs2(v)
			}
		}
	}

	for i := 0; i < n; i++ {
		if !suspicious[i] && !visited[i] {
			dfs2(i)
		}
	}

	ans := []int{}
	for i := 0; i < n; i++ {
		if !suspicious[i] {
			ans = append(ans, i)
		}
	}

	return ans
}
